web/app/fr/gospeak/web/auth/domain/SocialProfile.scala

```diff
@@ -0,0 +1,48 @@
+package fr.gospeak.web.auth.domain
+
+import com.mohiva.play.silhouette.api.LoginInfo
+import com.mohiva.play.silhouette.impl.providers.CommonSocialProfile
+import fr.gospeak.core.domain.User
+import fr.gospeak.libs.scalautils.StringUtils
+import fr.gospeak.libs.scalautils.domain.{Avatar, CustomException, EmailAddress, SocialProvider, Url}
+
+case class SocialProfile(loginInfo: LoginInfo,
+                         firstName: Option[String],
+                         lastName: Option[String],
+                         fullName: Option[String],
+                         email: Option[String],
+                         avatarURL: Option[String]) {
+  def toUserData: Either[CustomException, User.Data] = {
+    for {
+      socialProvider <- SocialProvider.from(loginInfo.providerID)
+      source <- socialProvider.toSource
+      //FIXME slug by default ??
+      slug <- User.Slug.from(StringUtils.slugify(firstName.getOrElse("")))
```

Comment:
should check if the slug is not already used, can be done in the UserRepo when creating the user with SocialProfile